sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_common/_response_handlers.py

```diff
@@ -23,34 +23,29 @@
 # IN THE SOFTWARE.
 #
 # --------------------------------------------------------------------------
-from typing import Any, Dict
-
-from azure.core.pipeline import PipelineResponse
-
-from .._generated.models import SchemaId as InternalSchemaId
 from ._schema import SchemaProperties, Schema
 
 
-def _parse_response_schema_id(pipeline_response, deserialized, response_headers):  # pylint: disable=unused-argument
-    # type: (PipelineResponse, InternalSchemaId, Dict[str, Any]) -> SchemaProperties
-    """
-
-    :param pipeline_response:
-    :param deserialized:
-    :param response_headers:
-    :return:
-    """
-    return SchemaProperties(schema_id=deserialized.id, **response_headers)
+def _parse_schema_properties(response):
+    return {
+        'location': response.headers.get('location'),
+        'location_by_id': response.headers.get('schema-id-location'),
+        'schema_id': response.headers.get('schema-id'),
+        'serialization_type': response.headers.get('serialization-type'),
+        'version': int(response.headers.get('schema-version'))
+    }
 
 
-def _parse_response_schema(pipeline_response, deserialized, response_headers):  # pylint: disable=unused-argument
-    # type: (PipelineResponse, str, Dict[str, Any]) -> Schema
-    """
+def _parse_response_schema_id(response):
+    properties_dict = _parse_schema_properties(response)
+    properties_dict['schema_id'] = response.json()["id"]
+    return SchemaProperties(
```

Comment:
🤔 it seems a little strange to me that _parse_schema_properties returns a dict and _parse_response_schema_id returns SchemaProperties (like the `get_schema_id` method that returned a `SchemaProperties` object). Is there a reason to not have `_parse_schema_properties` return a `SchemaProperties` object and `_parse_response_schema_id` return just the `schema_id` string?

Reply:
(especially since `_parse_response_schema` returns a `Schema` object)

Reply:
I renamed `_parse_schema_properties` -> `_parse_schema_properties_dict`
and `_parse_response_schema_id` -> `_parse_response_schema_properties`

does it look better now?

Reply:
yep, that looks good to me!